tests/unit/timing/strategies/test_user_centric_rate.py

```
@@ -304,6 +306,151 @@ def test_scale_down_suppresses_new_and_replacement_users(self) -> None:
+        assert gaps_1 != gaps_2
+
+    def test_missing_median_for_sampled_distribution_raises(self) -> None:
+        with pytest.raises(ValueError, match="user_centric_gap_median"):
+            _make_gap_strategy(distribution=UserCentricGapDistribution.LOGNORMAL)
+
+    def test_adaptive_rescale_below_median_raises_with_computed_mean(self) -> None:
+        strategy = _make_gap_strategy(
+            distribution=UserCentricGapDistribution.WEIBULL, median=0.3
+        )
+        # set_target_users(2) shrinks the mean gap to 2 / 10.0 = 0.2s < median.
+        with pytest.raises(ValueError, match="0.2"):
```

Ruff (0.15.21)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)